FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/EkmViewModel.kt

```diff
@@ -52,15 +56,29 @@ class EkmViewModel(application: Application) : BaseAndroidViewModel(application)
         val ekmPrivateResult = repository.getPrivateKeysViaEkm(
           context = context,
           ekmUrl = orgRules.keyManagerUrl
-            ?: throw java.lang.IllegalArgumentException("key_manager_url is empty"),
+            ?: throw IllegalArgumentException("key_manager_url is empty"),
           tokenId = tokenId
         )
 
         if (ekmPrivateResult.data?.privateKeys?.isEmpty() == true) {
-          throw java.lang.IllegalStateException(context.getString(R.string.no_prv_keys_ask_admin))
+          throw IllegalStateException(context.getString(R.string.no_prv_keys_ask_admin))
         }
 
-        ekmLiveData.value = ekmPrivateResult
+        val combinedSource = requireNotNull(
+          ekmPrivateResult.data?.privateKeys?.map { it.decryptedPrivateKey }
+            ?.joinToString(separator = "\n"))
```

Comment:
At this stage, we should check that all keys were fully decrypted when we fetched them. If any is encrypted at all, that's an unexpected error, we should throw an exception.

Reply:
> At this stage, we should check that all keys were fully decrypted when we fetched them. If any is encrypted at all, that's an unexpected error, we should throw an exception.

Got it. Will do. Also need to update the chart in #1168 to be clear

Reply:
I've added changes and updated the chart